### DISCUSSION TITLE
High cpu usage from diskspace plugin

### DISCUSSION THREAD

**Discussion by adrianbattiston** [netdata/netdata#15129] [Q&A]
Our 6 core linux system keeps getting a netdata process using 100% of a core. In the netdata monitoring section, it seems like the culprit is the diskspace plugin (see picture). This high cpu usage is usually accompanied by gsd-housekeeping (a gnome filesystem service) also eating up a bunch of cpu. I'm not sure if one is triggered by the other, but I'm curious if anyone knows why one might be triggering the other, or how to dial back the netdata diskspace plugin's cpu usage? It's a useful plugin for us - is my best bet to just dial back the update rate or are there other options?

<img width="938" alt="NetData diskspace usage" src="https://github.com/netdata/netdata/assets/9126513/117aa0c7-be65-49f0-9899-9e9f5a3de2f0">

**Comment by MrZammler** [comment]
Hi @adrianbattiston 

Can you tell us please the netdata version you're using? Have you noticed if this behavior started recently, i.e. after an update? Also, could look into `error.log` and `collectors.log` during that time?

**Comment by ilyam8** [comment]
Hi, @adrianbattiston. `diskspace` plugin is simple, the only things it does are:

 - periodically read `/proc/self/mountinfo`.
 - call `statvfs()` for every (not filtered out) mountpoint.

And the chart you shared is diskspace workers busy time, not CPU usage. 

- Can you share all diskspace workers chart? 
- Do you see increased CPU usage by `netdata` in Applications->cpu?
